lib/translator/common.h

```diff
@@ -15,14 +15,61 @@
 #ifndef LIB_TRANSLATOR_COMMON_H
 #define LIB_TRANSLATOR_COMMON_H
 
+#include "absl/types/span.h"
+
+#include "lib/scsi_defs.h"
+#include "third_party/spdk_defs/nvme_defs.h"
+
 namespace translator {
 
-enum class StatusCode { kSuccess, kInvalidInput, kNoTranslation, kFailure };
+enum class ApiStatus { kSuccess, kFailure };
+
+enum class StatusCode {
+  kSuccess,
+  kUninitialized,
+  kInvalidInput,
+  kNoTranslation,
+  kFailure
+};
 
 void DebugLog(const char* format, ...);
 
 void SetDebugCallback(void (*callback)(const char*));
 
+// Max consecutive pages required by NVMe PRP list is 512
+void* AllocPages(uint16_t count);
+
+void DeallocPages(void* pages_ptr, uint16_t count);
+
+void SetPageCallbacks(void* (*alloc_callback)(uint16_t),
```

Comment:
The definition uses the name `SetAllocPageCallbacks` but this prototype uses the name `SetPageCallbacks`. This mismatch will produce linker errors.